### DISCUSSION TITLE
One Week long using the mobile app only

### DISCUSSION THREAD

**Discussion by enahum** [mattermost/community#6] [General Ideas]
One of the big thing about messaging is the ability to communicate. In today's world we learned to rely on our mobile devices to get things done, specially to communicate with other people.

At Mattermost we spend a long time building features that would ease the path for communication to a lot of people but our focus has been while using our computers and not so much our mobile devices.

One thing is for sure, we have a great set of engineers, designers, support staff, sales staff, marketing and so much more, but more importantly we have a great community behind Mattermost. The idea is to have all this great people use our mobile apps to communicate and then post challenges and missing features that makes their life harder instead of easier when using the mobile app.

One we have this list of things, let's try as a whole, to help improve our own experiences using the app, so we are able to achieve and improve the way we all communicate.

Let's use this discussion to keep track of everything, so feel free to write your experience, the things you love, the things that you want to see, etc..

**Comment by esethna** [comment]
YES! Please everyone feel free to add comments or new discussions here for stuff that you'd like to see us improve

**Comment by gigawhitlocks** [comment]
In the image picker on Android, you have to select each individual image separately from the gallery. You can upload 5 at a time to Mattermost, of course, but in other applications you can select many images on the image selection screen. For Mattermost, you have to go back to Mattermost after selecting each image.

Not really a bug, but there's room for improvement there.

**Reply by esethna** [reply]
Yep, totally annoying. We have a ticket for an improved image picker: https://mattermost.atlassian.net/browse/MM-21167

**Reply by enahum** [reply]
already merged!!

**Reply by esethna** [reply]
@enahum really? The multiselect image selector as per the designs? Or are you referring to the gallery updates? 

Do you have a link to the PR?

**Reply by enahum** [reply]
https://github.com/mattermost/mattermost-mobile/pull/5474

**Reply by enahum** [reply]
not the pictures where the keyboard is, just the ability to select multiple at once

**Reply by gigawhitlocks** [reply]
Wow! Good stuff @enahum , I definitely didn't know there was already a _completed ticket_ for this!

**Comment by thefactremains** [comment]
Hey guys! I might be able to contribute some stuff here :)

I actually did this exercise of only using the mobile app not too long ago and I looked up my notes (which seem to still be relevant)

The most frustrating thing for me:
- **I cannot invite another user to the server from the mobile app**

This is a big issue for in-person community building. Anytime I want to invite someone I meet face to face, I have to either:
a) Create a reminder for myself to invite them later from my laptop (which means I cannot connect with the person right away and make sure they got in)
b) Take out my laptop to invite them (assuming I had it with me)

Some other minor issues:
- cannot change password from mobile app
- cannot change language from mobile app

**Reply by gigawhitlocks** [reply]
As an addition to this, the onboarding flow doesn't work (last I checked) on mobile either, so even if you go through the trouble of getting them a link (somehow), they can't follow the link successfully until they're back on their computer.

**Reply by esethna** [reply]
Thanks Dennis! Thanks for these!! Agree we need better parity in these areas

**Comment by thefactremains** [comment]
Ohhh, I almost forgot about this one. 

Remembering previously accessed server URLs would be super useful (at least until multi-server is out).

**Reply by esethna** [reply]
Ticket for this! https://mattermost.atlassian.net/browse/MM-19703

Although I hope multi-server support is not too far out ;)

**Comment by bbodenmiller** [comment]
Trying to read a channel with 300+ unread messages (MatterCon) is nearly impossible as the scrolling starts to get very jumpy.

**Comment by koox00** [comment]
I don't know if this exists but I'd love to be able to subscribe to more than one server simultaneously!

**Reply by enahum** [reply]
This is in progress

**Reply by esethna** [reply]
Yes this is part of Mobile v2, infrastructure in-progress!

**Comment by marianunez** [comment]
Would love to see my current drafts sync between mobile and desktop apps.

**Reply by esethna** [reply]
Upvote to this. This happens to me all the time where I start composing on mobile and then sit at my desk only to realize I have to finish the message on mobile if I don't want to loose my draft

**Reply by esethna** [reply]
Related idea, but not a replacement for syncing drafts in the channel context: https://github.com/mattermost/community/discussions/63

**Comment by chetanyakan** [comment]
System Admin Console should be available

**Reply by enahum** [reply]
@chetanyakan would you want the system admin console to be included or would it be better if it has a specialized app for it ?

**Reply by chetanyakan** [reply]
From a user's perspective, it might be easier to have it in the same app so that you don't have to install and authenticate in 2 different apps.

@enahum Is there an advantage of using/creating multiple apps?

**Reply by enahum** [reply]
@chetanyakan we could re-use the same authentication token stored in the keychain as having apps by the same author they can share data.

As for a different app, I think it reaches just the desired audience and does not add complexity and stuff to the current one.

Also the fact that I would expect the system console to match all the supported channels it seems that it would be way easier to maintain and keep backwards compatible that way.

Is just my opinion. That said I can also see how it can be a bit of a pain to have to install two different apps, although they do have two very different use cases.

**Reply by chetanyakan** [reply]
Yes, I can see how it may be easier implementation-wise to have 2 different apps.

**Comment by Willyfrog** [comment]
I'd totally do this but with an iPad and keyboard.

I don't own one anymore, but I used to read 
Mattermost in the morning in one.

**Reply by enahum** [reply]
but you could do it with your phone, right?

**Reply by esethna** [reply]
We should do this exercise for tablet too haha

**Reply by enahum** [reply]
1000000000% I believe that is a total must

**Reply by Willyfrog** [reply]
i could, but the screen is not so nice! :p

but yeah, if you organize either one, count me in! (although I should get an ipad again)
